channels_redis/core.py

```diff
@@ -191,6 +192,12 @@ def _setup_encryption(self, symmetric_encryption_keys):
                 # If we were the last out, stop the receive loop
                 if self.receive_count == 0:
                     self.receive_loop_task.cancel()
+                    # as tasks dont get cancelled immediately we will await the task
+                    # in order to make sure it gets cancelled properly
+                    try:
+                        await self.receive_loop_task
+                    except CancelledError:
+                        pass
```

Comment:
as with a task , it does not get cancelled immediately after calling cancel, CancelledError Need sto be raised for that task to cancel. 
[](https://stackoverflow.com/questions/40897428/please-explain-task-was-destroyed-but-it-is-pending)
Due to this when the loop stops after the tests, the task was pending. 
Awaiting it so as to raise the CancelledError and get the task cancelled before the receive loop completes.